projects/find/find.py

```diff
@@ -0,0 +1,38 @@
+import os
+import sys
+
+"""This method is created to find all files and directories, given a specific path. Optionally, your can use -name flag
+to search only the files that match to a specific name"""
+
+
+def find(path, option=None, filename=None):
+    """this method uses a bfs approach to resolve to the problem of finding all files and directories."""
+    if not os.path.exists(path):
+        return "Path doesn't exist or wrong: please try again"
+    queue = []
+    queue.append(path)
+    paths_list = []
+    while len(queue) > 0:
+        queue_element = queue.pop(0)
+        adj_list = os.listdir(queue_element)
+        if len(adj_list) != 0:
+            for adj in adj_list:
+                if filename is not None and option == "-name":
```

Comment:
Also, if option is -name can filename be None?
It looks to me that if I pass the -name option with a None filename you should throw me an error, and you should do so at the beginning. If the function is not called correctly why bother starting the while at all?

This is also something that looks like parsing logic to me, and it doesn't belong to this function at all. Why does the function need to know that there are options and one of them is called -name? You should think of the command line interface as the frontend of your application, and this function as the backend. Changes in the frontend should not impact (ideally) the backend.